### PR TITLE
Throw on invalid payload

### PR DESCRIPTION
**Pull request by mcollina**
Ref https://github.com/nearform/fast-jwt/pull/30#issuecomment-674795968

### CODE REVIEW

**Comment by mcollina** [comment]
cc @panva . Given this, we should probably remove the `json` option in `decode` and the if that caused this problem in the first place. I'll open an issue for that as it's a bigger change than the time I have available.

File: src/decoder.js [hunk 1]
```diff
@@ -43,8 +52,9 @@ function decode({ json, complete }, token) {
   }
 }
 
-module.exports = function createDecoder(options) {
-  const { json, complete } = { json: true, ...options }
+module.exports = function createDecoder(options = {}) {
+  const json = !(options.json === false)
```
Comment (dgonzalez) [inline src/decoder.js:56]: any reason for negation compare to `false`. Just my curiosity

Comment (mcollina) [reply src/decoder.js:56]: eslint was complaining.

File: src/decoder.js [hunk 2]
```diff
@@ -28,6 +28,15 @@ function decode({ json, complete }, token) {
     // Parse payload if needed
     if (json === true || header.typ === 'JWT') {
       payload = JSON.parse(payload)
+
+      // https://tools.ietf.org/html/rfc7519#section-7.2
+      //
+      // 10.  Verify that the resulting octet sequence is a UTF-8-encoded
+      //      representation of a completely valid JSON object conforming to
+      //      RFC 7159 [RFC7159]; let the JWT Claims Set be this JSON object.
+      if (typeof payload !== 'object') {
```
Comment (panva) [inline src/decoder.js:37]: typeof null (which is valid top level json as well) will pass this

Comment (mcollina) [reply src/decoder.js]: Added the check, good spot!

**Pull request merged**